### ISSUE TITLE
buttons and leds

### ISSUE THREAD

**Issue by benokio** [kentforth/Control-Surface-Frontend#1]
Good day. I have a question. I saw
   Indicate each bank with led # 136 and I have the same problem with 3 leds. I do not use a multiplexer. In the sketch of the tttapa there are {2, 3, 5}, // button pins, in this situation only the buttons of 2,3 work. If I change the order of {3, 5, 2}, // button pins - only 3 and 5 work. How did you solve this problem?

**Comment by kentforth** [comment]
Hi! Did you read all comments in this issue? May be you have buttons that are not normally closed. or may be buttons are just have bad wiring. Try another buttons or jumpers to connect it

**Comment by benokio** [comment]
I read the topic, but I see that you opened it, but did not give an answer whether you solved the problem. Santta then comments on connecting buttons, but he does it with a multiplexer. Tttapa tells him: You can invert all digitalRead results in the ManyButtonsSelector source code.
Once you find the answer in what others comment ok, but I can not understand that this is so. Thank you for your answer. (although it is a question). You also ask 100 questions and as I see tttapa does not ask you if you are absolutely aware of what you are doing!

**Comment by kentforth** [comment]
You can try invert buttons. May be your buttons are not normally closed. Like here:
https://github.com/tttapa/Control-Surface/issues/336

or Here:
https://github.com/tttapa/Control-Surface/issues/62

**Comment by benokio** [comment]
thanks a lot for the reply. I'm starting the experiments. :)

**Comment by benokio** [comment]
Yes, my buttons are normally open. The problem is that everything is soldered and I really can't figure out how to invert the buttons. I use two examples from Control Surface  : "Program Changer",  and the following example from Pieter:

#include <Control_Surface.h>

USBMIDI_Interface midi;

template <setting_t N>
class CCSelection : public Selectable<N> {
  public:
    CCSelection(const Array<MIDIAddress, N> &addresses)
      : addresses{addresses} {}


    void select(setting_t setting) override {
      setting = Selectable<N>::validateSetting(setting);
      Control_Surface.sendCC(addresses[setting], 0x7F);
    }

  private:
    Array<MIDIAddress, N> addresses;
};

CCSelection<3> cc = {{27, 30, 31}};


ManyButtonsSelectorLEDs<3> selector = {
  cc,           // what to select
  {39, 41, 43},    // button pins
  {5, 4, 6}, // led pins
};


void setup() {
  Control_Surface.begin();
}


void loop() {
  Control_Surface.loop();
}
